Blockzilla/IntroViewController.swift

```diff
@@ -26,6 +26,104 @@ struct IntroViewControllerUX {
     static let FadeDuration = 0.25
 }
 
+protocol PageControlDelegate {
+    func selectedIndex(_ index:Int)
+}
+
+class PageControl: NSObject {
+    
+    var currentPage = 0
+    var numberOfPages = 0
+    var hidesForSinglePage = false
+    
+    var stack : UIStackView? = nil
+    var delegate: PageControlDelegate?
+    
+    init(numberOfPages: Int) {
+        super.init()
+
+        var buttonArray: [UIButton]!
+        
+        // Ensure we have at least one button
+        if numberOfPages == 0 {
+            return
+        }
+        
+        for _ in 0..<numberOfPages {
+            let button = UIButton()
+            button.setImage(drawDot(), for: .normal)
+            buttonArray.append(button)
+        }
+        
+        // Style the StackView
+        stack = UIStackView(arrangedSubviews: buttonArray)
+        stack?.spacing = 20
+        stack?.distribution = .fillEqually
+        stack?.alignment = .center
+        
+        // Enable the buttons to be tapped to switch to next page
+        buttonArray.forEach({ button in
+            button.addTarget(self, action: #selector(selected(sender:)), for: .touchUpInside)
+        })
+        
+        // Default to the first page
+        selectIndex(0)
+    }
+    
+    func selectIndex(_ index:Int){
+        for button in (stack?.subviews)! as! [UIButton] {
```

Comment:
Force-unwraps are not permitted except in rare circumstances (sometimes it is syntactically difficult to avoid having to do this). Use `if let` or `guard else { return}`. LMK if you need more info on this.

On Firefox iOS we follow https://github.com/raywenderlich/swift-style-guide, and there is a section on optionals https://github.com/raywenderlich/swift-style-guide#optionals, but that doesn't provide guidance on this. (Note to self: check Focus iOS code standard)